File: src/main/java/de/kilobyte22/voc/api/common/networking/BidirectionalDataStream.java
package de.kilobyte22.voc.api.common.networking;

import java.io.*;

public class BidirectionalDataStream implements Closeable {

    private final DataInputStream in;
    private final DataOutputStream out;

    public BidirectionalDataStream(DataInputStream in, DataOutputStream out) {
        this.in = in;
        this.out = out;
    }

    public DataInput getInput() {
        return in;
    }

    public DataOutput getOutput() {
        return out;
    }

    public void close() throws IOException {
        in.close();
        out.close();
    }

    public void flush() throws IOException {
        out.flush();
    }
}

File: src/main/scala/de/kilobyte22/voc/api/common/networking/BidirectionalDataStream.scala
package de.kilobyte22.voc.api.common.networking

import java.io._

class BidirectionalDataStream(in: DataInputStream, out: DataOutputStream) extends Closeable {

  def getInput: DataInput = in

  def getOutput: DataOutput = out

  def close {
    in.close
    out.close
  }

  def flush {
    out.flush
  }
}